GCP/deploy.sh

```diff
@@ -2,78 +2,100 @@
 ZONE="europe-west1-b"
 PROJECT="default"
-DEFAULT_PROJECT=`gcloud config list --format 'value(core.project)'`
-DEFAULT_ZONE=`gcloud config list --format 'value(compute.zone)'`
+DEFAULT_PROJECT=$(gcloud config list --format 'value(core.project)')
+DEFAULT_ZONE=$(gcloud config list --format 'value(compute.zone)')
 ZONE="${DEFAULT_ZONE:-$ZONE}"
 PROJECT="${DEFAULT_PROJECT:-$PROJECT}"
 CLUSTER_NAME="${1:-k3s}"
+FW_RULE_NAME="allow-tcp-6443"
+
+#Put your ssh private key path here if not default
+privkeypath="/home/nedsi/.ssh/julien-rsa"
+
+#Default ssh private key path
+if [ -f "/home/$USER/.ssh/id_rsa" ]; then
+  default_privkeypath=$(/home/"$USER"/.ssh/id_rsa)
+fi
+privkeypath=${default_privkeypath:-$privkeypath}
 
 if [ "$2" = "delete" ]
 then
-  gcloud compute instances delete $CLUSTER_NAME-master $CLUSTER_NAME-worker1 $CLUSTER_NAME-worker2 $CLUSTER_NAME-worker3
+  gcloud compute instances delete "$CLUSTER_NAME"-master "$CLUSTER_NAME"-worker1 "$CLUSTER_NAME"-worker2 "$CLUSTER_NAME"-worker3
   echo "Cluster $CLUSTER_NAME deleted"
+  gcloud compute firewall-rules describe "$FW_RULE_NAME" --no-user-output-enabled --quiet >/dev/null 
+  echo "Firewall rule $FW_RULE_NAME deleted"
   exit 0
 fi
 
 echo "----- K3S GO!!! -----"
 
-# If you want to provide ssh key add this line:
-#--metadata=ssh-keys="[USER_AS_IN_SSH_KEY]:[YOUR_SSH_KEY_HERE]" \
-gcloud compute --project=$PROJECT instances create $CLUSTER_NAME-master \
---zone=$ZONE \
---machine-type=n1-standard-2 \
---tags=k3smaster,k3s-$CLUSTER_NAME \
---subnet=default \
---network-tier=PREMIUM \
---maintenance-policy=MIGRATE \
---image-family=ubuntu-minimal-2004-lts \
---image-project=ubuntu-os-cloud \
---no-user-output-enabled >/dev/null &
-
-gcloud compute --project=$PROJECT instances create $CLUSTER_NAME-worker1 $CLUSTER_NAME-worker2 $CLUSTER_NAME-worker3 \
---zone=$ZONE \
---machine-type=n1-standard-2 \
---tags=k3s-$CLUSTER_NAME \
---subnet=default \
---network-tier=PREMIUM \
---maintenance-policy=MIGRATE \
---image-family=ubuntu-minimal-2004-lts \
---image-project=ubuntu-os-cloud \
---no-user-output-enabled >/dev/null &
+# If you want to provide ssh key add this line under the gcloud command:
+gcloud compute --project="$PROJECT" instances create "$CLUSTER_NAME"-master \
+  --metadata=ssh-keys="julien.sudan:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDEauZPOCeD7ygrp1Z/tML12Q0gwlYg7WC2EIqJAgT9AD1t4cXCZ+4yMPpdZBbuRvCCaBXNz69VhSxqKXahtmOvlRkLRoZ03U5Xt60Le+oEI/FZFTKCRoAlMhEhlm+lVoeo736YusNe3N9bIqnK5KiKDCxBV1d4hiMlEHeeh1DA1thl/UYew7zNHn7Sibv06WZC7LmkYvtX2knCr3WJbufHCzwm9GaUYqEoZJJ7wxOjAIti2gHLXRgNQCqWoxa7E5XKJ9cqgTGMonGP4+Nw2NBzOwI2MeVwC1foTx1m89EBIUiClNc7KRYiKGCgBV1fGe+NSxVbIgHAfrWFxZro1olfGg+dvPWxPasT/wkGGvi7MbyFGjA1yKgQgWwnXOhs5BWle8Lm5kTs202T7YpCZ+QNE6mltuLCRF+/hk2hG6QErKBAz+KGD2btCn2rM725lX3/I4/NAQIslPlhdQrz+4kFXTjO5gi9jWXsBBtkvxR5WObq4bvaDNdBZ/Y4fVkOamM= julien.sudan@container-solutions.com" \
```

Comment:
Probably best if you take your SSH key out ;) Also, I never explicitly specified it here because I (and I believe most people) have their SSH keys added to their Google Cloud account and then it's automatically inherited for each VM. So in most cases, this won't be necessary, but it's definitely good to have the option to specify it. But nevertheless, please remove your key from here

Reply:
Mistake again. I added it here as, for some reason, it wasn't working without it on my side. Might have been a wrong setup from me tho.